Floating button

So1ene opened the pull request:
Jira Issue: https://jira.corp.adobe.com/browse/DOTCOM-63512

- Floating button on mobile & tablet only
- You can import the function `createFloatingButton($a)` from `blocks/floating-button/floating-button.js`. The function will create a floating button using the `$a` CTA link. I used this function for compatibility with the old version of floating button that already existed in the `fullsize-columns` block.
- You can also make a floating-button by creating a block called Floating Button and inside the table insert a CTA

- The arrow lottie animation scrolls you down to the `template-list` block, `layouts` block or `.steps-highlight-container` if one of them exists, or the 3rd `section-wrapper` if it doesn't exist.
- Scrolling past those sections changes the button to hide the arrow lottie animation, and adds a slight white gradient below the button

- `a.button.gradient` Added a new button class called `gradient` with hover animation, using this as a basis: https://jira.corp.adobe.com/secure/attachment/8405636/image-2022-05-20-09-32-32-417.png

XD design: https://xd.adobe.com/view/6d0702f1-edb1-4051-97c6-c31a0b92f4a6-3854/screen/c6e439a7-e194-4634-b127-196661f644d2/ Winning version: Ch2 (Gradient color)

### Test URLs: ###
- Before: https://main--express-website--adobe.hlx3.page/express/create/flyer
- After: https://floating-button--express-website--Webistry-Development.hlx3.page/express/create/flyer?lighthouse=on

Examples of pages that use the floating button:
[/express/create/logo](https://floating-button--express-website--Webistry-Development.hlx3.page/express/create/logo?lighthouse=on)
[/express/create/flyer](https://floating-button--express-website--Webistry-Development.hlx3.page/express/create/flyer?lighthouse=on)
[/express/create/banner](https://floating-button--express-website--Webistry-Development.hlx3.page/express/create/banner?lighthouse=on)
[/express/create/advertisement](https://floating-button--express-website--Webistry-Development.hlx3.page/express/create/advertisement?lighthouse=on)
[/express/create/poster](https://floating-button--express-website--Webistry-Development.hlx3.page/express/create/poster?lighthouse=on)
[/express/create/business-card](https://floating-button--express-website--Webistry-Development.hlx3.page/express/create/business-card?lighthouse=on)
[/express/create/thumbnail/youtube](https://floating-button--express-website--Webistry-Development.hlx3.page/express/create/thumbnail/youtube?lighthouse=on)
[/express/feature/image/resize](https://floating-button--express-website--Webistry-Development.hlx3.page/express/feature/image/resize?lighthouse=on)
[/express/feature/image/remove-background](https://floating-button--express-website--Webistry-Development.hlx3.page/express/feature/image/remove-background?lighthouse=on)
[/express/feature/video/resize](https://floating-button--express-website--Webistry-Development.hlx3.page/express/feature/video/resize?lighthouse=on)
[/express/feature/video/merge](https://floating-button--express-website--Webistry-Development.hlx3.page/express/feature/video/merge?lighthouse=on)
[/express/feature/video/convert/gif-to-mp4](https://floating-button--express-website--Webistry-Development.hlx3.page/express/feature/video/convert/gif-to-mp4?lighthouse=on)

rofe commented:
This looks really great already, @So1ene. A few comments:
- the example URL is a page with fullsize column block. This is the special case where we "auto-block" the `floating-button` block. In this case, the original CTA from the columns block should be hidden or removed to avoid confusion on which button to click
- can you also add a draft page where the `floating-button` is added to the page explicitly via authoring?
- while the page is loading with empty cache, I can see the floating button appear at the top of the page, before it snaps into place at the bottom. Can you make it appear at the bottom or keep it hidden until it's ready to be displayed?

So1ene commented:
> This looks really great already, @So1ene. A few comments:
> 
> * the example URL is a page with ...

I believe I fixed all the feedback, including the ones from the Jira ticket:
-  `gradient` button now using this animation: [gradient css.rtf](https://jira.corp.adobe.com/secure/attachment/8415167/8415167_gradient+css.rtf)

- Hides all the other CTA's on the page that have the same href to avoid confusion (except for the one in the purple banner)

- Await loading the css before showing to avoid the floating-button appearing at the top of the page

- Here is the draft page where the floating-button is added to the page explicitly via authoring:

Before: https://main--express-website--adobe.hlx3.page/drafts/solene/feature
After: https://floating-button--express-website--webistry-development.hlx3.page/express/create/flyer?lighthouse=on

rofe commented:
@So1ene very nice! 
- I can still see the gradient button appear at the top when simulating slow 3G:
   <img width="389" alt="image" src="https://user-images.githubusercontent.com/1609742/170938238-75acf728-68ba-4b50-a5c3-301bca4dbeac.png">
   You probably need to hide the container using a rule in `styles.css` and add an `appear` class when done.
- I think the automatic hiding of other CTAs based on the target URL could be confusing for authors. I'd probably omit that feature for now and first see if there will be cases with duplicate CTAs.
- Can you update your fork and merge main into your `floating-button` branch? The `how-to-carousel` changes seem unrelated.

So1ene commented:
- Rebased and updated to remove code from `main` branch that was conflicting with `ccx0033` branch

Updated the shared/carsouel.js file for real seamless infinite scrolling

Before:
- https://ccx0033--express-website--adobe.hlx3.page/express/
After:
- https://floating-button--express-website--webistry-development.hlx3.page/express/

rofe commented on express/blocks/floating-button/floating-button.js:
```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
```
```suggestion
 * Copyright 2022 Adobe. All rights reserved.
```

rofe commented on express/styles/styles.css:
```diff
@@ -339,6 +339,36 @@ a.button.dark.reverse:any-link:focus {
+a.button.gradient:any-link:focus {
+  color: var(--color-white);
+  border-color: transparent;
+  background: linear-gradient(320deg, #7C84F3, #FF4DD2, #FF993B, #FF4DD2, #7C84F3, #FF4DD2,#FF993B );
```
Should we store this in a var?

So1ene replied:
Up to you, not sure if we will use it again...

rofe commented:
@So1ene I think this PR should also remove the scroll button from `template-list.sixcols` as it is now integrated with `floating-button`.

rofe commented:
Thanks @So1ene . This is ready to go from my POV!
